package-coverage/generator/generator.go

```diff
@@ -78,10 +91,35 @@ type Generator struct {
 
 	// Tags is arguments passed to the go test runner
 	Tags string
+
+	// Concurrency controls how many tests can be run concurrently.  Default is `runtime.NumCPU()`
+	Concurrency int
 }
 
 func (g *Generator) do(paths []string) {
+	jobsCh := make(chan string, len(paths))
+	wg := &sync.WaitGroup{}
+
+	// create workers
+	for index := 1; index <= g.Concurrency; index++ {
+		wg.Add(1)
```

Comment:
I would place this before the `for`, and do `wg.add(g.Concurrency)` instead, to prevent a race of `Done` with `Add`.

Reply:
Good call.  Feels cleaner too